sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/pipeline.py

```diff
@@ -97,7 +97,7 @@ def settings(self) -> PipelineJobSettings:
 
     @settings.setter
     def settings(self, value):
-        if value is not None and not isinstance(value, PipelineJobSettings):
+        if value and not isinstance(value, PipelineJobSettings):
             raise TypeError("settings must be PipelineJobSettings or dict but got {}".format(type(value)))
```

Comment:
Seems we don't allow dict here?

Reply:
Yes. As `settings` is not recommended in v2, we should raise error here if we pass a dict with value. In this bug, we pass an empty dict, therefore we can roughly modify this line to fix this bug.